DigitalLearningSolutions.Web/Controllers/LearningPortalController/SelfAssessment.cs

```diff
@@ -261,11 +286,33 @@ public IActionResult SupervisorComments(int selfAssessmentId, int competencyNumb
                 PreviousCompetencyNumber = Math.Max(competencies.Count(), 1),
                 NumberOfOptionalCompetencies = optionalCompetencies.Count(),
                 SupervisorSignOffs = supervisorSignOffs,
+                SearchViewModel = new SearchSelfAssessmentOvervieviewViewModel(searchModel?.SearchText, assessment.Id, vocabulary, searchModel?.AppliedFilters)                
             };
             ViewBag.SupervisorSelfAssessmentReview = assessment.SupervisorSelfAssessmentReview;
             return View("SelfAssessments/SelfAssessmentOverview", model);
         }
 
+        private List<Competency> FilterCompetencies(List<Competency> competencies, SearchSelfAssessmentOvervieviewViewModel search)
+        {
+            var searchText = search?.SearchText?.Trim() ?? string.Empty;
+            List<Competency> filteredCompetencies = null;
+            if (searchText.Length > 0 || search?.AppliedFilters.Count() > 0)
+            {
+                var wordsInSearchText = searchText.Split().Where(w => w != string.Empty);
+                var filters = search.AppliedFilters.Select(f => Enum.Parse<SelfAssessmentResponseStatus>(f.FilterValue));
+                filteredCompetencies = (from c in competencies
+                        let searchTextMatchesGroup = wordsInSearchText.Any(w => c.CompetencyGroup?.Contains(w, StringComparison.CurrentCultureIgnoreCase) ?? false)
+                        let searchTextMatchesDescription = wordsInSearchText.Any(w => c.Description?.Contains(w, StringComparison.CurrentCultureIgnoreCase) ?? false)
+                        let responseStatus = c.AssessmentQuestions.FirstOrDefault()?.ResultId == null ? SelfAssessmentResponseStatus.NotYetResponded
+                            : c.Verified != null ? SelfAssessmentResponseStatus.SelfAssessed

```

Comment:
`: c.Verified != null ? SelfAssessmentResponseStatus.SelfAssessed` should be `: c.Verified == null ? SelfAssessmentResponseStatus.SelfAssessed`